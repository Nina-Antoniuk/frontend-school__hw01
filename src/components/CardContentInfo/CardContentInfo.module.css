.video {
  display: block;
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 20px;
  max-width: 100%;
}

@media screen and (min-width: 768px) {
  .video {
    max-width: 600px;
    max-height: 500px;
  }
}

.hashtags {
  display: inline-block;
}

.placeholder {
  display: block;
  margin-left: auto;
  margin-right: auto;
}
/*# sourceMappingURL=CardContentInfo.module.css.map */
