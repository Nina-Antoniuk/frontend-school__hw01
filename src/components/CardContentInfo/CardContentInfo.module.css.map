{
    "version": 3,
    "mappings": "AAEA,AAAA,MAAM,CAAC;EACL,OAAO,EAAE,KAAK;EACd,WAAW,EAAE,IAAI;EACjB,YAAY,EAAE,IAAI;EAClB,aAAa,EAAE,IAAI;EACnB,SAAS,EAAE,IAAI;CAMhB;;AAJC,MAAM,CAAC,MAAM,MAAM,SAAS,EAAE,KAAK;EAPrC,AAAA,MAAM,CAAC;IAQH,SAAS,EAAE,KAAK;IAChB,UAAU,EAAE,KAAK;GAEpB;;;AAED,AAAA,SAAS,CAAC;EACR,OAAO,EAAE,YAAY;CACtB;;AAED,AAAA,YAAY,CAAC;EACX,OAAO,EAAE,KAAK;EACd,WAAW,EAAE,IAAI;EACjB,YAAY,EAAE,IAAI;CACnB",
    "sources": [
        "CardContentInfo.module.scss",
        "../../shared/scss/variables.scss"
    ],
    "names": [],
    "file": "CardContentInfo.module.css"
}