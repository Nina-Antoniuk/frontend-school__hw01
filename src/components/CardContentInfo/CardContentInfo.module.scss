@import '../../shared//scss/variables.scss';

.video {
  display: block;
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 20px;
  max-width: 100%;

  @media screen and(min-width: $breakpoint-min-tablet) {
    max-width: 600px;
    max-height: 500px;
  }
}

.hashtags {
  display: inline-block;
}

.placeholder {
  display: block;
  margin-left: auto;
  margin-right: auto;
}
