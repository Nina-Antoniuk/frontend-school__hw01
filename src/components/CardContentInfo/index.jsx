import React from 'react';
import CardContent from '@mui/material/CardContent';
import Typography from '@mui/material/Typography';
import { PropTypes } from 'prop-types';
import styles from './CardContentInfo.module.scss';

const CardContentInfo = function CardContentInfo({ video, desc }) {
  return (
    <CardContent>
      {video ? (
        <video
          className={styles.video}
          src={video}
          controls
          autoPlay
          muted
          loop
          preload="auto"
        />
      ) : (
        <img
          className={styles.placeholder}
          src="https://via.placeholder.com/250x300.png?text=video+is+not+available"
          alt="placeholder"
        />
      )}

      <Typography variant="body2" color="text.secondary">
        {desc}
      </Typography>
    </CardContent>
  );
};

CardContentInfo.defaultProps = {
  video: '',
  desc: '',
};

CardContentInfo.propTypes = {
  video: PropTypes.string,
  desc: PropTypes.string,
};

export default CardContentInfo;
