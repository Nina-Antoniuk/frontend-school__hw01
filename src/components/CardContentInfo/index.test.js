import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import CardContentInfo from './index';

let container = null;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('condition-dependent component render check', () => {
  test('render CardContentInfo with fake link', () => {
    const fakeProps = {
      video: 'https://www.youtube.com/watch?v=Ex2j4oIweDk',
      hashtags: [
        { id: '123', name: 'jest' },
        { id: '123', name: 'enzyme' },
      ],
      desc: 'about unit tests',
    };
    act(() => {
      render(<CardContentInfo {...fakeProps} />, container);
    });
    expect(container).toMatchSnapshot();
  });

  test('render CardContentInfo without fake link', () => {
    const fakeProps = {
      video: '',
      hashtags: [
        { id: '123', name: 'jest' },
        { id: '123', name: 'enzyme' },
      ],
      desc: 'about unit tests',
    };
    act(() => {
      render(<CardContentInfo {...fakeProps} />, container);
    });
    expect(container).toMatchSnapshot();
  });
});
