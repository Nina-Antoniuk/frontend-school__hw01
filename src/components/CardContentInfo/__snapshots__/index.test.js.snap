// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`condition-dependent component render check render CardContentInfo with fake link 1`] = `
<div>
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <video
      autoplay=""
      class="video"
      controls=""
      loop=""
      preload="auto"
      src="https://www.youtube.com/watch?v=Ex2j4oIweDk"
    />
    <p
      class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
    >
      about unit tests
    </p>
  </div>
</div>
`;

exports[`condition-dependent component render check render CardContentInfo without fake link 1`] = `
<div>
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <img
      alt="placeholder"
      class="placeholder"
      src="https://via.placeholder.com/250x300.png?text=video+is+not+available"
    />
    <p
      class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
    >
      about unit tests
    </p>
  </div>
</div>
`;
